### PR TITLE
Add publishRate config for RabbitMQ Scaler

### PR DESCRIPTION
**Pull request by rwkarg**
Adding `publishRate` config for RabbitMQ scaler. This is mutually exclusive with the existing `queueLength` config.
The use is for scaling higher throughput scenarios where there is little queue length backlog or the queue length doesn't adequately correlate to the number of instances required to match the workload.

### Checklist

- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [X] Tests have been added
- [-] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [ ] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [ ] Changelog has been updated

Fixes #

### CODE REVIEW

File: pkg/scalers/rabbitmq_scaler.go [hunk 1]
```diff
@@ -282,19 +335,29 @@ func (s *rabbitMQScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
...
 
 // GetMetrics returns value for a supported metric and an error if there is a problem getting the metric
 func (s *rabbitMQScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
-	messages, err := s.getQueueMessages()
+	messages, publishRate, err := s.getQueueStatus()
 	if err != nil {
 		return []external_metrics.ExternalMetricValue{}, fmt.Errorf("error inspecting rabbitMQ: %s", err)
 	}
 
+	if s.metadata.queueLength > 0 {
```
Comment (zroubalik) [inline pkg/scalers/rabbitmq_scaler.go:349]: Could you please change it to not duplicate the code, ie:
```go
if s.metadata.queueLength > 0 {
  metricValue = *resource.NewQuantity(int64(messages), resource.DecimalSI)
}else{
  metricValue = resource.NewMilliQuantity(int64(publishRate*1000), resource.DecimalSI)
}

metric := external_metrics.ExternalMetricValue{
			MetricName: metricName,
			Value:      metricValue,
			Timestamp:  metav1.Now(),
		}

return append([]external_metrics.ExternalMetricValue{}, metric), nil
```

File: pkg/scalers/rabbitmq_scaler.go [hunk 2]
```diff
@@ -269,10 +303,29 @@ func (s *rabbitMQScaler) getQueueInfoViaHTTP() (*queueInfo, error) {
 
 // GetMetricSpecForScaling returns the MetricSpec for the Horizontal Pod Autoscaler
 func (s *rabbitMQScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
-	targetMetricValue := resource.NewQuantity(int64(s.metadata.queueLength), resource.DecimalSI)
+	if s.metadata.queueLength > 0 {
```
Comment (zroubalik) [inline pkg/scalers/rabbitmq_scaler.go:306]: Could you please reformat the code to remove the duplication, the same way it is suggested in `GetMetrics()`

**Comment by rwkarg** [comment]
This is working in our clusters, both with a `publishRate` trigger and both `publishRate` and `queueLength` triggers (and the existing `queueLength` only ScaledObjects continue to work as before).

I'll let this bake overnight and then submit another PR with the correct DCO comment in the commit.

**Comment by rwkarg** [comment]
Closing as this is replaced with https://github.com/kedacore/keda/pull/1653

**Pull request closed without merging**